internal/core/metadata/v2/controller/http/deviceprofile.go

```diff
@@ -134,3 +134,32 @@ func (dc *DeviceProfileController) AddDeviceProfileByYaml(w http.ResponseWriter,
 	// Encode and send the resp body as JSON format
 	pkg.Encode(addDeviceProfileResponse, w, lc)
 }
+
+func (dc *DeviceProfileController) GetDeviceProfileByName(w http.ResponseWriter, r *http.Request) {
+	lc := container.LoggingClientFrom(dc.dic.Get)
+	ctx := r.Context()
+	correlationId := correlation.FromContext(ctx)
+
+	// URL parameters
+	vars := mux.Vars(r)
+	name := vars[v2.Name]
+
+	var response interface{}
+	var statusCode int
+
+	deviceProfile, err := application.GetDeviceProfileByName(name, ctx, dc.dic)
+	if err != nil {
+		if errors.Kind(err) != errors.KindEntityDoesNotExist {
+			lc.Error(err.Error(), clients.CorrelationHeader, correlationId)
+		}
+		lc.Debug(err.DebugMessages(), clients.CorrelationHeader, correlationId)
+		response = commonDTO.NewBaseResponse("", err.Message(), err.Code())
+		statusCode = err.Code()
+	} else {
+		response = responseDTO.NewDeviceProfileResponse("", "", http.StatusOK, deviceProfile)
+		statusCode = http.StatusOK
+	}
+
+	utils.WriteHttpHeader(w, ctx, statusCode)
```

Comment:
please also modify other functions to use `utils.WriteHttpHeader(w, ctx, statusCode)`

Reply:
Modifed.